commit 0acc05caf7518cd0d63ab02bfa036455add02346
Author: Nobuyoshi Nakada <nobu@ruby-lang.org>
Date:   Tue Feb 2 08:33:08 2021 +0900

    Add -fdeclspec to CFLAGS instead of CC

diff --git a/configure.ac b/configure.ac
index 1206e90153..fc6523408b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -550,11 +550,16 @@ AS_IF([test x"${RPATHFLAG}" = x], [
     rpathflag=`echo "$RPATHFLAG" | sed 's/%.*//'`
 ])
 
-RUBY_TRY_CFLAGS(-fdeclspec, [
-  RUBY_APPEND_OPTIONS(CC, -fdeclspec)
-  RUBY_APPEND_OPTIONS(MJIT_CC, -fdeclspec)
+RUBY_TRY_CFLAGS(-fdeclspec, [fdeclspec=yes], [fdeclspec=no])
+AS_IF([test "$fdeclspec" = yes], [
+    RUBY_APPEND_OPTIONS(CFLAGS, -fdeclspec)
+    RUBY_APPEND_OPTIONS(cflags, -fdeclspec)
+    RUBY_APPEND_OPTIONS(orig_cflags, -fdeclspec)
+])
+RUBY_TRY_CXXFLAGS(-fdeclspec, [fdeclspec=yes], [fdeclspec=no])
+AS_IF([test "$fdeclspec" = yes], [
+    RUBY_APPEND_OPTIONS(CXXFLAGS, -fdeclspec)
 ])
-RUBY_TRY_CXXFLAGS(-fdeclspec, [RUBY_APPEND_OPTIONS(CXX, -fdeclspec)])
 
 AS_CASE([$RUBY_PATCHLEVEL], [-*],
 	[RUBY_DEVEL=yes], [RUBY_DEVEL=no])
